Sources/SemanticAnalyzer/SemanticError.swift

```diff
@@ -108,6 +108,12 @@ extension Diagnostic {
   static func renderCapabilityGroup(_ capabilities: [CallerCapability]) -> String {
     return "\(capabilities.map({ $0.name }).joined(separator: ", "))"
   }
+
+  static func recursiveStructs(_ identifier: Identifier, _ root: PropertyInformation) -> Diagnostic {
+    let note = Diagnostic(severity: .note, sourceLocation: root.sourceLocation, message: "'\(identifier.name)' is reused here")
+    return Diagnostic(severity: .error, sourceLocation: identifier.sourceLocation, message: "Recursive declaration inside of '\(identifier.name)'", notes: [note])
```

Comment:
What about `Error: Declaration of recursive struct '...'`, `Note: State property '...' of type '...' refers to enclosing type  '...'`?